demo/angular-editable-text.js

```diff
@@ -10,31 +10,32 @@
 
 })();
 
-/**
- * Created by Gabriel Grinberg on 6/13/14.
- */
+/*
+   Created by Gabriel Grinberg on 6/13/14.
+   */
 
-(function () {
+(function() {
   'use strict';
-  angular.module('gg.editableText')
-    .directive('editableText', ['EditableTextHelper', function (EditableTextHelper) {
+  angular.module('gg.editableText', ['puElasticInput'])
+    .directive('editableText', ['$timeout', 'EditableTextHelper', function($timeout, EditableTextHelper) {
       return {
         scope: {
           editableText: '=',
           editMode: '=',
           placeholder: '@',
-          onChange: '&'
+          onChange: '&',
+          onReject: '&',
         },
         transclude: true,
-        template: '<span ng-class="{\'is-placeholder\': placeholder && !editingValue}">' +
-        '<input ng-show="isEditing" ng-blur="isEditing=false;" ng-keypress="($event.which === 13) && (isEditing = false)" ng-model="editingValue" placeholder="{{placeholder}}"/>' +
-        '<span ng-hide="isEditing || isWorking" class="original-text" tabindex="0" ng-click="isEditing=true" ng-focus="isEditing=true;">{{placeholder ? (editingValue ? editingValue : placeholder) : editingValue}}</span>' +
-        '<span ng-hide="isEditing" ng-transclude></span>' +
-        '<span ng-show="isWorking" class="' + EditableTextHelper.workingClassName + '">' + EditableTextHelper.workingText + '</span>' +
-        '</span>',
-        link: function (scope, elem, attrs) {
-          var input = elem.find('input'),
-            lastValue;
+        template:
+          '<span ng-class="{\'is-placeholder\': placeholder && !editingValue}" ng-style="{\'max-width\': \'inherit\'}" >' +
+            '<input ng-blur="onInputBlur()" ng-click="onInputClick()" ng-keydown="onKeyPress($event)" ng-model="editingValue" placeholder="{{placeholder}}"' +
+              'type="text" pu-elastic-input pu-elastic-input-minwidth="auto" pu-elastic-input-maxwidth="inherit" />' +
+          // '<span ng-show="isWorking" class="' + EditableTextHelper.workingClassName + '">' + EditableTextHelper.workingText + '</span>' +
```

Comment:
I think if you just change the existing `ng-show/hide` checks to a simple function that checks `isWorking && EditableTextHelper.workingText.length` then you can continue to support this feature by defaulting workingText to an empty string.